lightning/kv/sql2kv.go

```diff
@@ -69,13 +69,16 @@ func (kvcodec *TableKVEncoder) Encode(
 	}
 
 	for i, col := range cols {
-		if j := columnPermutation[i]; j >= 0 {
+		if j := columnPermutation[i]; j >= 0 && j < len(row) {
 			value, err = table.CastValue(kvcodec.se, row[j], col.ToInfo())
 			if err == nil {
 				value, err = col.HandleBadNull(value, kvcodec.se.vars.StmtCtx)
 			}
+		} else if mysql.HasAutoIncrementFlag(col.Flag) {
+			// we still need a conversion, e.g. to catch overflow with a TINYINT column.
+			value, err = table.CastValue(kvcodec.se, types.NewIntDatum(rowID), col.ToInfo())
 		} else {
-			value, err = table.GetColOriginDefaultValue(kvcodec.se, col.ToInfo())
+			value, err = table.GetColDefaultValue(kvcodec.se, col.ToInfo())
```

Comment:
not handle some cases like `current_timestamp` anymore?

Reply:
`CURRENT_TIMESTAMP` does work, it currently fetches the actual current time 🤔. (Before this PR it was always NULL.) Gonna change to a fixed timestamp stored in the checkpoint to ensure determinism in a future PR.